dbfit-java/snowflake/src/main/java/dbfit/environment/SnowflakeEnvironment.java

```diff
@@ -0,0 +1,211 @@
+        return getConnectionString(dataSource) + "?db=" + database;
+    }
+
+    public Map<String, DbParameterAccessor> getAllColumns(String tableOrViewName)
+            throws SQLException {
+        String[] qualifiers = NameNormaliser.normaliseName(tableOrViewName)
+                .split("\\.");
+        String qry = " select column_name, data_type, character_maximum_length "
+                + "from information_schema.columns where ";
+        if (qualifiers.length == 2) {
+            qry += " lower(table_schema)=? and lower(table_name)=? ";
+        } else {
+            qry += " (table_schema=database() and lower(table_name)=?)";
+        }
+        qry += " order by ordinal_position";
+        return readColumnsFromDb(qualifiers, qry);
+    }
+
+    private Map<String, DbParameterAccessor> readColumnsFromDb(
+            String[] parametersForColumnQuery, String query) throws SQLException {
+        try (PreparedStatement dc = currentConnection.prepareStatement(query)) {
+            for (int i = 0; i < parametersForColumnQuery.length; i++) {
+                dc.setString(i + 1,
+                        NameNormaliser.normaliseName(parametersForColumnQuery[i]));
+            }
+            ResultSet rs = dc.executeQuery();
+            Map<String, DbParameterAccessor> columns = new HashMap<String, DbParameterAccessor>();
+            int position = 0;
+            while (rs.next()) {
+                String columnName = rs.getString(1);
+                if (columnName == null)
+                    columnName = "";
+                String dataType = rs.getString(2);
+                DbParameterAccessor dbp = createDbParameterAccessor(
+                        columnName,
+                        Direction.INPUT, getSqlType(dataType),
+                        getJavaClass(dataType), position++);
+                columns.put(NameNormaliser.normaliseName(columnName), dbp);
+            }
+            rs.close();
+            return columns;
+        }
+    }
+
+    // List interface has sequential search, so using list instead of array to
+    // map types
+    private static List<String> stringTypes = Arrays.asList(new String[] {
+            "VARCHAR", "CHAR", "TEXT", "STRING"});
+    private static List<String> booleanTypes = Arrays
+            .asList(new String[] { "BOOLEAN" });
+
+    private static List<String> decimalTypes = Arrays.asList(new String[] {"DECIMAL", "NUMBER", "NUMERIC",
+            "INT", "INTEGER", "TINYINT", "SMALLINT" , "BIGINT"});
+    private static List<String> timestampTypes = Arrays.asList(new String[] {
+            "DATETIME", "TIMESTAMP_LTZ", "TIMESTAMP_NTZ", "TIMESTAMP_TZ" });
+    private static List<String> dateTypes = Arrays.asList("DATE");
+    private static List<String> timeTypes = Arrays.asList("TIME");
+
+     private static List<String> variantTypes = Arrays.asList("VARIANT");
+
+    private static int getSqlType(String dataType) {
+        // todo:strip everything from first blank
```

Comment:
Is there anything pending TODO?

Reply:
That was a not for me earlier, I missed removing this.